### PR TITLE
Fix counterfactual SDK issues from bug bash

### PR DESCRIPTION
**Pull request by gaugup**
The PR addresses the following issues:-
- Improve error message for classification scenario
- Fix the breakage of feature_to_vary feature in counterfactual manager
- Add validation for feature names in perimitted_range in counterfactual manager
- Enabled tests for feature_to_vary and perimitted_range without validation.

### CODE REVIEW

File: responsibleai/tests/counterfactual/test_counterfactual_advanced_features.py [hunk 1]
```diff
@@ -47,19 +45,19 @@ def test_counterfactual_vary_features(
...
-        for feature_name in feature_names:
-            if not vary_all_features and feature_name != feature_names[0]:
-                assert np.all(
-                    cf_obj.cf_examples_list[0].final_cfs_df[feature_name] ==
-                    X_test.iloc[0:1][feature_name][0])
-            else:
-                assert np.any(
-                    cf_obj.cf_examples_list[0].final_cfs_df[feature_name] !=
-                    X_test.iloc[0:1][feature_name][0])
+        # TODO: The logic below needs to be made robust for gated tests
```
Comment (gregorybchris) [inline responsibleai/tests/counterfactual/test_counterfactual_advanced_features.py:48]: Could you use `np.isclose` in this PR?
https://numpy.org/doc/stable/reference/generated/numpy.isclose.html

File: responsibleai/responsibleai/_managers/counterfactual_manager.py [hunk 2]
```diff
@@ -116,13 +116,23 @@ def _add_counterfactual_config(self, new_counterfactual_config):
...
+                message = ("Feature names in permitted_range do "
+                           f"not exist in train data: {list(difference_set)}")
+                raise UserConfigValidationException(message)
 
         if self._task_type == ModelTask.CLASSIFICATION:
             if new_counterfactual_config.desired_class is None:
                 raise UserConfigValidationException(
                     'The desired_class attribute should be either \'{0}\''
-                    ' or the class value for classification scenarios.'.format(
+                    ' binary classification or the class value for '
```
Comment (gregorybchris) [inline responsibleai/responsibleai/_managers/counterfactual_manager.py:134]: *for binary classification

Comment (gaugup) [reply responsibleai/responsibleai/_managers/counterfactual_manager.py:134]: Done

**Review by gregorybchris — APPROVED** [review]

File: responsibleai/tests/counterfactual/test_counterfactual_advanced_features.py (hunk 1 above)
Comment (gregorybchris) [reply responsibleai/tests/counterfactual/test_counterfactual_advanced_features.py:48]: After discussion, this can be done in a follow-up

Comment (gaugup) [reply responsibleai/tests/counterfactual/test_counterfactual_advanced_features.py:48]: Opened issue https://github.com/microsoft/responsible-ai-widgets/issues/758 to track this.

**Pull request merged**